lib/Rex/Commands/Rsync.pm

```diff
@@ -161,7 +161,7 @@ sub sync {
     Rex::Logger::debug("Downloading $source -> $dest");
     push @rsync_cmd, "rsync -rl --verbose --stats $params ";
 
-    $source = $auth->{user} . "\@$servername:$source";
+    $source = $auth->{user} . "\@$servername:$source" if $auth;
```

Comment:
Regarding this and other checks of `$auth`:

- is it enough to check for truth?
- would it be better to check for "does `$auth` look like a HASH ref"?
- can the check be made earlier in the code once, instead of checking it in many places?